test/integration/master/kube_apiserver_test.go

```diff
@@ -190,6 +197,153 @@ func TestOpenAPIDelegationChainPlumbing(t *testing.T) {
+	defer server.TearDownFn()
+	kubeclient, err := kubernetes.NewForConfig(server.ClientConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	apiextensionsclient, err := apiextensionsclientset.NewForConfig(server.ClientConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	crdPath, exist, err := getOpenAPIPath(kubeclient, `/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/{name}`)
+	if err != nil {
+		t.Fatalf("unexpected error getting CRD OpenAPI path: %v", err)
+	}
+	if !exist {
+		t.Fatalf("unexpected error non existing CRD OpenAPI path")
+	}
+	crdDefinition, exist, err := getOpenAPIDefinition(kubeclient, `io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.CustomResourceDefinition`)
+	if err != nil {
+		t.Fatalf("unexpected error getting CRD OpenAPI definition: %v", err)
+	}
+	if !exist {
+		t.Fatalf("unexpected error non existing CRD OpenAPI definition")
+	}
+	// Create a CRD that overlaps OpenAPI path with the CRD API
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "customresourcedefinitions.apiextensions.k8s.io",
+			Annotations: map[string]string{"api-approved.kubernetes.io": "unapproved, test-only"},
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "apiextensions.k8s.io",
+			Version: "v1beta1",
+			Scope:   apiextensionsv1beta1.ClusterScoped,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:   "customresourcedefinitions",
+				Singular: "customresourcedefinition",
+				Kind:     "CustomResourceDefinition",
+				ListKind: "CustomResourceDefinitionList",
+			},
+			Validation: &apiextensionsv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
+					Type: "object",
+					Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+						"readonly": {Type: "boolean"},
+					},
+				},
+			},
+		},
+	}
+	etcd.CreateTestCRDs(t, apiextensionsclient, false, crd)
+	// Expect the CRD path to not change
+	if err := wait.PollImmediate(5*time.Second, 15*time.Second, func() (bool, error) {
+		path, _, err := getOpenAPIPath(kubeclient, `/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/{name}`)
+		if err != nil {
+			return false, err
+		}
+		if !cmp.Equal(path, crdPath, cmp.AllowUnexported(jsonreference.Ref{}), cmp.AllowUnexported(jsonpointer.Pointer{})) {
+			return false, fmt.Errorf("expected CRD OpenAPI path to not change, but got different results, diff:\n%s", cmp.Diff(path, crdPath, cmp.AllowUnexported(jsonreference.Ref{}), cmp.AllowUnexported(jsonpointer.Pointer{})))
+		}
+		return false, nil
+	}); err != nil && err != wait.ErrWaitTimeout {
+		t.Fatalf("failed to observe CRD OpenAPI path to not change: %v", err)
+	}
+	// Expect the orphan definition to be pruned
+	_, exist, err = getOpenAPIDefinition(kubeclient, `io.k8s.apiextensions.v1beta1.CustomResourceDefinition`)
```

Comment:
this is sensitive to changes in names (which we don't guarantee to stay the same). The test will keep working. Would rather put some identifier into the schema and just string search it in the JSON spec.